mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -1169,32 +1169,49 @@ public override bool IsDefined (Type attributeType, bool inherit)
 
 		public override object[] GetCustomAttributes (bool inherit)
 		{
-			return base.GetCustomAttributes (inherit);
+			return GetCustomAttributes (null, inherit);
 		}
 
 		public override object[] GetCustomAttributes (Type attributeType, bool inherit)
 		{
-			return base.GetCustomAttributes (attributeType, inherit);
+			if (cattrs == null || cattrs.Length == 0)
+				return new object [] {};
+
+			List<object> results = new List<object> ();
+			for (int i=0; i < cattrs.Length; i++) {
+				Type t = cattrs [i].Ctor.GetType ();
+
+				if (attributeType == null || attributeType.IsAssignableFrom (t))
```

Comment:
This ignores input arguments `inherit`

Reply:
@marek-safar `GetCustomAttributes(Type, Boolean inherit)` is [documented to ignore the `inherit` paremeter](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.emit.modulebuilder.getcustomattributes?view=netframework-4.7#System_Reflection_Emit_ModuleBuilder_GetCustomAttributes_System_Type_System_Boolean_).  Is there some way we could explicitly mark it as ignored in C#?

Reply:
Interesting, I think a test case in this case would be appropriate

Reply:
How does this work with SRE defined cattr? How does it behave in case of an open TypeBuilder? How does it behave in case of a closed TypeBuilder?

Reply:
I think .NET checks for unbaked types and throws

Reply:
I can do that.